File: v131016/TeeNew/Axis_LabelsFormat.pas
unit Axis_LabelsFormat;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, Series, TeeProcs, Chart;

type
  TAxisLabelsFormat = class(TBaseForm)
    Series1: TFastLineSeries;
    procedure Chart1GetAxisLabel(Sender: TChartAxis; Series: TChartSeries;
      ValueIndex: Integer; var LabelText: String);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TAxisLabelsFormat.Chart1GetAxisLabel(Sender: TChartAxis;
  Series: TChartSeries; ValueIndex: Integer; var LabelText: String);
var Num : Double;
begin
  Num:=StrToFloat(LabelText);

  if Sender=Chart1.Axes.Left then
  begin
    if Num>300 then Sender.LabelsFont.Color:=clRed
               else Sender.LabelsFont.Color:=clBlue;
  end
  else
  if Sender=Chart1.Axes.Bottom then
  begin
    if Num>12  then Sender.LabelsFont.Color:=clRed
               else Sender.LabelsFont.Color:=clBlue;
  end;
end;

initialization
  RegisterClass(TAxisLabelsFormat);
end.

File: TeeNew/Builder2009/Axis_LabelsFormat.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Axis_LabelsFormat.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TAxisLabelsFormat *AxisLabelsFormat;
//---------------------------------------------------------------------------
__fastcall TAxisLabelsFormat::TAxisLabelsFormat(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TAxisLabelsFormat::Chart1GetAxisLabel(TChartAxis *Sender,
      TChartSeries *Series, int ValueIndex, AnsiString &LabelText)
{
  double Num = StrToFloat(LabelText);
  if (Sender == Chart1->Axes->Left)
  {
    if (Num>300) Sender->LabelsFont->Color = clRed;
    else Sender->LabelsFont->Color = clBlue;
  }
  else if (Sender == Chart1->Axes->Bottom)
  {
    if (Num>12) Sender->LabelsFont->Color = clRed;
    else Sender->LabelsFont->Color = clBlue;
  }
}
//---------------------------------------------------------------------------

